webpack.development.js

```diff
@@ -0,0 +1,204 @@
+                    },
+                    ignoreBrowserslistConfig: true,
+                  },
+                ],
+                ...babelConfig.presets,
+              ],
+              plugins: [...babelConfig.plugins],
+              cacheCompression: false,
+              cacheDirectory: true,
+            },
+          },
+        ],
+      },
+      imageLoader('static/'),
+    ]
+  }
+}
+
+const legacyBabelConfig = [
+  '@babel/env',
+  {
+    bugfixes: true,
+    useBuiltIns: 'usage',
+    corejs: '3.14'
+  },
+]
+
+const legacyBabelConfigNodeModules = [
+  '@babel/env',
+  {
+    bugfixes: true,
+    useBuiltIns: 'usage',
+    corejs: '3.14',
+    modules: 'amd'
+  },
+]
+
+const modernBabelConfig = [
+  '@babel/env',
+  {
+    bugfixes: true,
+    targets: {
+      esmodules: true
+    }
+  }
+];
+
+const browserConfig = (isLegacy) => ({
+  module: {
+    rules: [
+      {
+        exclude: /node_modules/,
+        test: /\.(m?)(j|t)s(x?)/,
+        use: [
+          {
+            loader: 'babel-loader',
+            options: {
+              presets: [
+                isLegacy ? legacyBabelConfig : modernBabelConfig
+                ,
+                ...babelConfig.presets,
+              ],
+              plugins: [...babelConfig.plugins],
+              cacheCompression: false,
+              cacheDirectory: true,
+            },
+          },
+        ],
+      },
+      {
+        include: /node_modules/,
+        exclude: [
+          /node_modules[\\\/]core-js/,
+          /node_modules[\\\/]@babel/,
+          /node_modules[\\\/]webpack[\\\/]buildin/,
+        ],
+        test: /\.(m?)(j|t)s(x?)/,
+        use: [
+          {
+            loader: 'babel-loader',
+            options: {
+              presets: [
+                isLegacy ? legacyBabelConfigNodeModules : modernBabelConfig
+                ,
+                ...babelConfig.presets,
+              ],
+              plugins: [...babelConfig.plugins],
+              cacheCompression: false,
+              cacheDirectory: true,
+            },
+          },
+        ],
+      },
+      imageLoader('./')
+    ]
+  },
+  plugins: [
+    ...isLegacy ? [] : [new ForkTsCheckerWebpackPlugin({
+      async: true,
+      typescript: {
+        mode: 'write-references' // for better babel-loader perf.
+      }
+    })],
+    new ForkTsCheckerNotifierWebpackPlugin({ excludeWarnings: true, skipFirstNotification: true }),
+    new AssetsPlugin({
+      path: path.resolve(__dirname, 'build'),
+      filename: `${isLegacy ? 'legacy.' : ''}webpack-assets.json`
+    })
+  ]
+});
+
+const baseServerConfig = baseConfig[0];
+const baseBrowserLegacyConfig = baseConfig[1];
+const baseBrowserConfig = baseConfig[2];
+
+module.exports = [
+  {
+    ...baseServerConfig,
+    ...common,
+    ...serverConfig,
+  },
+  {
+    ...baseBrowserLegacyConfig,
+    ...common,
+    ...browserConfig(true),
+  },
+  {
+    ...baseBrowserConfig,
+    ...common,
+    ...browserConfig(false),
+  }
+]
```

Comment:
Using [`webpack-merge`](https://github.com/survivejs/webpack-merge) rather than spread should mean that you don't need to duplicate as much code, as it concatenates arrays and merges objects creating a new object.

It should mean that you can overwrite/add properties/values that you need, rather than having the whole object.

See `webpack.config.js` on how we use it there.

Reply:
pushed up a fix for this using mergeWithRules :)